app/graphql/types/topic_details_type.rb

```diff
@@ -0,0 +1,25 @@
+module Types
+  class TopicDetailsType < Types::BaseObject
+    field :id, ID, null: false
+    field :title, String, null: false
+    field :last_activity_at, GraphQL::Types::ISO8601DateTime, null: true
+    field :live_replies_count, Int, null: false
+    field :likes_count, Int, null: false
+    field :topic_category_id, ID, null: true
+    field :creator_name, String, null: true
+    field :created_at, GraphQL::Types::ISO8601DateTime, null: false
+
+    def creator_name
+      object.first_post.creator&.name
+    end
+
+    def likes_count
+      object.first_post.post_likes.count
+    end
+
+    def live_replies_count
+      object.live_replies.count
+    end
```

Comment:
@harigopal : I did try using `ActiveRecord::Precounter`` yesterday and removed it because of the high allocations. The precounter counts the replies of every topic that satisfies the main `topics` query in the resolver, though we are interested in 10 records handled by `Graph` pagination. The `SimilarTopicsResolver` however has a set limit of 5 in the resolver method itself, where we won't face this issue.

Reply:
Hmm. Understood.